fahrschein/src/main/java/org/zalando/fahrschein/AuthorizationBuilder.java

```diff
@@ -0,0 +1,66 @@
+package org.zalando.fahrschein;
+
+import org.zalando.fahrschein.domain.Authorization;
+import org.zalando.fahrschein.domain.Authorization.AuthorizationAttribute;
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static org.zalando.fahrschein.domain.Authorization.AuthorizationAttribute.ANYONE;
+
+/**
+ * Use factory method {@link AuthorizationBuilder#authorization()} to create
+ * an allow-all authorization object and use {@link #withAdmins} / {@link #withReaders}
+ * methods to override admins and/or readers lists.
+ */
+public class AuthorizationBuilder {
+
+    private final List<AuthorizationAttribute> admins;
+    private final List<AuthorizationAttribute> readers;
+
+    private AuthorizationBuilder(List<AuthorizationAttribute> admins, List<AuthorizationAttribute> readers) {
+        this.admins = admins;
+        this.readers = readers;
+    }
+
+    /**
+     * No restrictions by default, use {@link #withAdmins} / {@link #withReaders} methods to add them.
+     */
+    public static AuthorizationBuilder authorization() {
+        return new AuthorizationBuilder(singletonList(ANYONE), singletonList(ANYONE));
+    }
+
+    public AuthorizationBuilder withAdmins(List<AuthorizationAttribute> admins) {
+        return new AuthorizationBuilder(admins, readers);
+    }
+
+    public AuthorizationBuilder withReaders(List<AuthorizationAttribute> readers) {
+        return new AuthorizationBuilder(admins, readers);
+    }
+
+    public AuthorizationBuilder withAdmins(AuthorizationAttribute... admins) {
+        return new AuthorizationBuilder(asList(admins), readers);
+    }
+
+    public AuthorizationBuilder withReaders(AuthorizationAttribute... readers) {
+        return new AuthorizationBuilder(admins, asList(readers));
+    }
+
+    public AuthorizationBuilder addAdmin(String dataType, String value) {
+        List<AuthorizationAttribute> newAdmins = admins.stream().filter(aa -> !aa.equals(ANYONE)).collect(toList());
+        newAdmins.add(new AuthorizationAttribute(dataType, value));
+        return withAdmins(newAdmins);
+    }
+
+    public AuthorizationBuilder addReader(String dataType, String value) {
+        List<AuthorizationAttribute> newReaders = readers.stream().filter(aa -> !aa.equals(ANYONE)).collect(toList());;
```

Comment:
I'd do this in the getter.

1. Initialize the lists with an empty list
2. Add admin/reader by creating a copy of the current list + the new role
3. In getter
   - if list is empty, return `singletonList(ANYONE)`
   - else, return admins/readers

Reply:
If you don't want to make a full copy every time, I'd suggest to use https://github.com/GlenKPeterson/Paguro.

Reply:
`return withReaders(admins.append(new AuthorizationAttribute(..)));`

Reply:
Yeah, sounds like a good idea, let me try. I am not sure it will be more elegant than the current state, but we'll see.

Reply:
Implemented your proposal, see below. It got 4 lines bigger and arguably better. I'm fine with whatever approach, I think the one you suggested is also quite nice.

I don't wanna add any dependencies just for the sake of elementary collections mapping, because this is a library, that's used across many projects and dragging more dependencies to all of them sounds like a bad idea to me.